plugins/modules/ec2_metadata_facts.py

```diff
@@ -646,6 +653,13 @@ def fetch_session_token(self, uri_token):
             token_data = None
         return to_text(token_data)
 
+    def get_instance_tags(self, tag_keys):
+        tags = {}
+        for key in tag_keys:
+            value = self._fetch("{}/{}".format(self.uri_instance_tags, key))
```

Comment:
I believe this information should already be lurking in `self._data`, the extra fetches aren't needed.  We already spit this data out under `ansible_ec2_tags_instance_{key_name}`

Reply:
Good point, I'll update. I'm assuming we want to keep that data as is, along with the current `ansible_ec2_instance_tags_keys` return value.